### DISCUSSION TITLE
is there a way to have postgrest load auto_explain

### DISCUSSION THREAD

**Discussion by foxydevloper** [PostgREST/postgrest#2734] [Q&A] [answered]
i have my frontend app with loads of traffic going through postgrest to the database, but the database is hitting very high load because a query is taking a long time. i heard about auto_explain, but it doesn't seem like there's a way to get postgrest to load auto_explain in. I don't have the ability to modify session_preload_libraries so I can't figure out how to get postgrest to load in auto_explain. Or, if there's another way to debug queries that are causing huge load on my database

**Comment by steve-chavez** [comment, marked as answer]
You might be able to do it like:

```sql
ALTER ROLE authenticator SET session_preload_libraries = 'auto_explain';
```

Assuming `authenticator` is your connection role.

**Comment by gregwebs** [comment]
Here are some basic performance troubleshooting approaches: https://www.crunchydata.com/blog/exposing-postgres-performance-secrets.
